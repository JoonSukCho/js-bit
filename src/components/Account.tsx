import React from 'react';

const Account = ({ name, mbti }: { name: string; mbti: string }) => {
  return (
    <div>
      <h2>{name}</h2>
      <h2>{mbti}</h2>
      <h2>hellou</h2>
    </div>
  );
};

export default Account;
