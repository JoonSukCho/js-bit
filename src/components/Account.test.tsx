import Account from './Account';
import { render } from '@testing-library/react';

it('matches snapshot', () => {
  const utils = render(<Account name="chojs" mbti="isfj" />);

  // 컴포넌트를 수정했을 때 기존 스냅샷과 비교해
  // 원하는 방식으로 렌더링되는지 비교하는 테스트
  expect(utils.container).toMatchSnapshot();
});

it('shows the props correctly', () => {
  const utils = render(<Account name="chojs" mbti="isfj" />);

  // 특정 문구가 있는지 확인하는 테스트
  utils.getByText('chojs');
  utils.getByText('isfj');
});
