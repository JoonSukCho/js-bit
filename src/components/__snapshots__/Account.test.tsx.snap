// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div>
  <div>
    <h2>
      chojs
    </h2>
    <h2>
      isfj
    </h2>
    <h2>
      hellou
    </h2>
  </div>
</div>
`;
